218

kmartinchek opened the pull request:
Issue 218 for review

dgon32 commented on 09_employment/compute_metrics_wage_ratio_2022.do:
```diff
@@ -0,0 +1,168 @@
+save "wage_ratio_final_2022.dta", replace
+
+** add in county name and state name
+global crosswalk "C:\Users\KMartinchek\Documents\umf-2024\mobility-from-poverty\geographic-crosswalks\data"
```
need to move this up top so that users can update directories in one place

kmartinchek replied:
Moved.

dgon32 commented on 09_employment/compute_metrics_wage_ratio_2022.do:
```diff
@@ -0,0 +1,168 @@
+/* drop statewide obs */
+drop if _merge == 1
+
+/* only keep 1 adult, 2 children row */
```
comment with reason for this drop

kmartinchek replied:
Added

dgon32 commented on 09_employment/compute_metrics_wage_ratio_2022.do:
```diff
@@ -0,0 +1,168 @@
+
+gen new_ratio = string(average_to_living_wage_ratio)
+drop average_to_living_wage_ratio
+rename new_ratio average_to_living_wage_ratio
```
to be consistent with variable conventions in the readme, change to "ratio_average_to_living_wage"

kmartinchek replied:
Under issue, we renamed this to: ratio_living_wage. So I have updated this variable to have this name (see issue comments).

dgon32 commented on 09_employment/compute_metrics_wage_ratio_2022.do:
```diff
@@ -0,0 +1,168 @@
+/* rename final variable here */
+
+*rename ratio_average_to_living_wage average_to_living_wage_ratio
+rename wage_ratio_quality ratio_living_wage_quality
```
simplify code by naming this variable "ratio_living_wage_quality" when you initially create that variable in line 77

kmartinchek replied:
Simplified.

dgon32 commented on 09_employment/compute_metrics_wage_ratio_2022.do:
```diff
@@ -0,0 +1,168 @@
+
+/***** save living wage as .dta *****/
+cd `wages'
+import delimited using "mit-living-wage-2022.csv", clear // while in code review, please note that I did not change this, as updated data required a fee and there was a no-scraping notice
```
reminder to remove this comment when final

kmartinchek replied:
Removed.

dgon32 commented on 09_employment/compute_metrics_wage_ratio_2022.do:
```diff
@@ -0,0 +1,168 @@
+per discussion with Greg, >= 30 is 1, <30 is 3 */
+gen wage_ratio_quality = 1 if annualaverageestablishmentcount >= 30 & annualaverageestablishmentcount != .
+replace wage_ratio_quality = 3 if annualaverageestablishmentcount < 30 & annualaverageestablishmentcount != .
+replace wage_ratio_quality = . if annualaverageestablishmentcount == .
```
if the quality flag is set to missing, should the metric also be set to missing?

kmartinchek replied:
Lines 96-97 replaces missing ratios with NA and then sets quality flag to 3, which addresses this-- this was how quality flags were determined in prior code and there were no notes to update this.

dgon32 commented on 09_employment/compute_metrics_wage_ratio_2022_subgroup_new.do:
```diff
@@ -0,0 +1,200 @@
+
+/***** save living wage as .dta *****/
+cd `wages'
+import delimited using "mit-living-wage-2022.csv", clear // while in code review, please note that I did not change this, as updated data required a fee and there was a no-scraping notice
```
reminder to remove comment from internal review

kmartinchek replied:
Removed

dgon32 commented on 09_employment/compute_metrics_wage_ratio_2022_subgroup_new.do:
```diff
@@ -0,0 +1,200 @@
+save "wage_ratio_final_2022_subgroup.dta",replace
+
+** add in county name and state name
+global crosswalk "C:\Users\KMartinchek\Documents\umf-2024\mobility-from-poverty\geographic-crosswalks\data"
```
move up so directory paths are all in one place

kmartinchek replied:
Moved.

dgon32 commented on 09_employment/compute_metrics_wage_ratio_2022_subgroup_new.do:
```diff
@@ -0,0 +1,200 @@
+
+gen new_ratio = string(average_to_living_wage_ratio)
+drop average_to_living_wage_ratio
+rename new_ratio average_to_living_wage_ratio
```
rename variable so it starts with "ratio"

kmartinchek replied:
Changed (see comment above re: new naming convention discussed in issue)

dgon32 commented on 09_employment/compute_metrics_wage_ratio_2022_subgroup_new.do:
```diff
@@ -0,0 +1,200 @@
+per discussion with Greg, >= 30 is 1, <30 is 3 */
+gen wage_ratio_quality = 1 if annualaverageestablishmentcount >= 30 & annualaverageestablishmentcount != .
+replace wage_ratio_quality = 3 if annualaverageestablishmentcount < 30 & annualaverageestablishmentcount != .
+replace wage_ratio_quality = . if annualaverageestablishmentcount == .
```
same comment as other do file to check whether the metric should also be set to missing

kmartinchek replied:
See comment above.

dgon32 commented on 09_employment/compute_metrics_wage_ratio_2022_subgroup_new.do:
```diff
@@ -0,0 +1,200 @@
+/* drop statewide obs */
+drop if _merge == 1
+
+/* only keep 1 adult, 2 children row */
```
delete commented out code

kmartinchek replied:
Removed

dgon32 commented on 09_employment/compute_metrics_wage_ratio_2022_subgroup_new.do:
```diff
@@ -0,0 +1,200 @@
+
+/* rename final variable here */
+
+*rename ratio_average_to_living_wage average_to_living_wage_ratio
```
name variables with the final names when you create them to simplify this code

kmartinchek replied:
Simplified.

dgon32 commented on 09_employment/compute_metrics_wage_ratio_2022_subgroup_new.do:
```diff
@@ -0,0 +1,200 @@
+order year state county state_name county_name subgroup_type subgroup  average_to_living_wage_ratio ratio_living_wage_quality
+
+export delimited using metrics_wage_ratio_2022_subgroup.csv, replace
+
```
sort file by year, state, and county -- currently out of order

kmartinchek replied:
The "order" command orders them properly-- and when I open the csv, I see the correct order-- so I am not sure this is an issue.

dgon32 reviewed (changes requested):
left suggested changes in comments

kmartinchek commented on 09_employment/compute_metrics_wage_ratio_2022.do:
```diff
@@ -0,0 +1,168 @@
+
+keep state county year average_to_living_wage_ratio wage_ratio_quality
+
+/* rename final variable here */
```
Deleted.

kmartinchek commented:
I updated the code and pushed to branch 218.

dgon32 commented on 09_employment/compute_metrics_wage_ratio_2022.do:
```diff
@@ -0,0 +1,160 @@
+
+
+/***** update these directories *****/
+global raw "C:\Users\KMartinchek\Documents\umf-2024\mobility-from-poverty\09_employment"
```
updated code looks fine but need to update the macros in both do files from globals to locals, otherwise the directories aren't called and code doesn't run

kmartinchek replied:
Changed and re-pushed!

dgon32 reviewed (changes requested):
updated code looked fine, but the directories need to be changed from global to locals otherwise the directories don't work properly. once those are updated code is good to go.

dgon32 commented on 09_employment/~compute_metrics_wage_ratio_2022.do.stswp:
```diff
@@ -118,7 +118,7 @@ export delimited using metrics_wage_ratio_2022.csv, replace
 
 ** add in county name and state name
-import delimited using "$crosswalk\county-populations.csv", stringcols(2 3 4 5) varnames(1) clear
+import delimited using "`crosswalk\county-populations.csv", stringcols(2 3 4 5) varnames(1) clear
```
this is the last change I promise, everything else worked ok -- there is a quote missing at the end of crosswalk (should be `crosswalk')

dgon32 reviewed (changes requested):
just one tiny change

kmartinchek commented:
Hmm I am quite confused because when I look at branch 218 on GitHub and my local, the quotes ` and ' are there for all locals. I looked and tried to re-save and commit but it tells me the working tree is clean.

I think you are looking at the temp file (do.stswp) instead of the final .do file and that could be the error.

awunderground commented:
@kmartinchek Can we delete the temp file? Is it needed in the GitHub repo?

kmartinchek commented:
Removed temp files.

awunderground reviewed (changes requested):
The quality flag is missing where there are non-missing metrics.

dgon32 reviewed (changes requested):
For the subgroup file, the number of subgroups is still not the same across counties (total number of subgroup categories should be 7,  but there are counties with fewer than that). Also, per Claudia, the values for subgroup and subgroup_type should not be missing, even if there isn't data for a particular county/industry. Also recommend moving the ssc install line up and instructing users to uncomment that line if they don't have the distinct command installed.

kmartinchek commented:
To respond:
-I moved up the ssc install to top of do file
-When I run count if missing(subgroup) and count if missing(subgroup_type), there are no missing values and tab also shows no missingness
-The final csv had 25,144 observations, or 8 observations (1 for overall and 7 industries) for each of the 3.143 counties. I also added code to check that this is the case for all counties.

Can you give a little more detail on the files you were reviewing? From my end, it seems that these errors are not present.

dgon32 reviewed (approved):
not sure why Kassie's earlier edits didn't pull correctly earlier this morning, but double checked and everything looks fine now. counts for the subgroup file look correct.
